app/src/test/java/org/oppia/android/app/parser/StringToFractionParserTest.kt

```diff
@@ -73,99 +73,217 @@ class StringToFractionParserTest {
   }
 
   @Test
   fun testParser_realTimeError_noNumerator_returnsInvalidFormat() {
     val error = stringToFractionParser.getRealTimeAnswerError("/3")
-      .getErrorMessageFromStringRes(context)
-    assertThat(error).isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.INVALID_FORMAT)
+
+    val errorMessage = error.getErrorMessageFromStringRes(context)
+    assertThat(errorMessage)
+      .isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
   }
 
   @Test
   fun testParser_realTimeError_severalSlashes_returnsInvalidFormat() {
     val error = stringToFractionParser.getRealTimeAnswerError("1/3/8")
-      .getErrorMessageFromStringRes(context)
-    assertThat(error).isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.INVALID_FORMAT)
+
+    val errorMessage = error.getErrorMessageFromStringRes(context)
+    assertThat(errorMessage)
+      .isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
   }
 
   @Test
   fun testParser_realTimeError_severalDashes_returnsInvalidFormat() {
     val error = stringToFractionParser.getRealTimeAnswerError("-1/-3")
-      .getErrorMessageFromStringRes(context)
-    assertThat(error).isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.INVALID_FORMAT)
+
+    val errorMessage = error.getErrorMessageFromStringRes(context)
+    assertThat(errorMessage)
+      .isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+  }
+
+  @Test
+  fun testParser_parseFraction_divisionByZero() {
+    val parseFraction = stringToFractionParser.parseFraction("8/0")
+    val parseFractionFromString = stringToFractionParser.parseFractionFromString("8/0")
+    val constructedFraction = Fraction.newBuilder().apply {
+      numerator = 8
+    }.build()
+    assertThat(parseFractionFromString).isEqualTo(constructedFraction)
+    assertThat(parseFraction).isEqualTo(constructedFraction)
   }
 
   @Test
-  fun testParser_parseFractionFromString_incorrectFraction_throwsException() {
+  fun testParser_parseFraction_invalidFormat() {
+    val parseFraction = stringToFractionParser.parseFraction("7 1/2 4/5")
+    assertThat(parseFraction).isEqualTo(null)
+
     val exception = assertThrows(IllegalArgumentException::class) {
-      stringToFractionParser.parseFractionFromString("abc")
+      stringToFractionParser.parseFractionFromString("7 1/2 4/5")
     }
-    assertThat(exception)
-      .hasMessageThat()
-      .contains("Incorrectly formatted fraction: abc")
+    assertThat(exception).hasMessageThat().contains("Incorrectly formatted fraction: 7 1/2 4/5")
   }
 
   @Test
-  fun testParser_parseFraction_incorrectFraction_returnsNull() {
+  fun testParser_parseFraction_invalidChars() {
```

Comment:
Nit: test names should follow the format: ``testAction_withCondition_andCondition_hasOutcome``. Please update this & other test names that don't match.